server/Assignments_filler/sheetFiller.py

```diff
@@ -0,0 +1,237 @@
+                #pass
+
+        #if needed, add "recruit"
+        if rank == 4:
+            if localComment == "":
+                localComment = "Recruit"
+                print ("empty comment")
+            elif "Recruit" in localComment:
+                print ("comment contains recruit")
+                #pass
+            else:
+                print ("Recruit added to comment")
+                localComment = "Recruit, " + localComment
+            
+        #if no roles were filled, use "Rifleman"
+        if subject == "(No subject)":
+            subject = "Rifleman"
+
+        data = [name, subject, localComment]
+        writeRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+        writeSheet(service, data, writeRange, 3)
+        row = row + 1
+    
+    #delete leftovers
+    logging.info("Deleting leftovers...")
+    data = ["", "", ""]
+    checkRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+    NextCell = readSheet(service, checkRange, 3)
+    while NextCell != "":
+        writeSheet(service, data, checkRange, 3)
+        row = row + 1
+        checkRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+        NextCell = readSheet(service, checkRange, 3)
+    
+    logging.info("Successfully filled!")
+    
+    checker(service, contractName, nid, numberOfComments)
+
+def main():
+    #Setting up the API service
+    #region
+    creds = None
+    # The file token.json stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first time.
+    if os.path.exists('token.json'):
+        creds = Credentials.from_authorized_user_file('token.json', SCOPES)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                'credentials.json', SCOPES)
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open('token.json', 'w') as token:
+            token.write(creds.to_json())
+
+    service = build('sheets', 'v4', credentials=creds, cache_discovery=False)
+    readRange = TEAM_ASSINGMENT_TAB+RANGE_READ
+    
+    #read new contract name from DB
+    writtenName = readSheet(service, readRange)
+    query = ("SELECT `nid`,`title` FROM drupal.node where title like \"%%%s%%\"" % (writtenName))
+    contractList = get_from_db(query)
+    nid = contractList[0][0]
+    contractName = contractList[0][1]
+    query = ("SELECT `created` FROM drupal.comment where nid=%s ORDER BY created DESC" % (nid))
+    commentList = get_from_db(query)
+    numberOfComments = len(commentList)
+    
+    checker(service, contractName, nid, numberOfComments)
+
+if __name__ == '__main__':
+    main()
```

Comment:
Missing empty line at end of file (EOF). This is suggested for Git diffs to work properly and reduce merge conflicts.